### DISCUSSION TITLE
Combining json

### DISCUSSION THREAD

**Discussion by prashasti-k** [josdejong/mathjs#3258] [Q&A]
There is a union of arrays but how can I combine objects or add a key-value to an object using expressions?
I want something like 
`math.evaluate('merge(obj, { newKey: newValue }))', { obj: {a:1}, newValue: 3 })`
So that this replicates the spread operator as well like `{ ...obj, newKey: newValue}`

**Comment by josdejong** [comment]
You can write a `merge` function yourself and `import` it in your mathjs instance. There is indeed no built in solution for this (yet). If there is enough interest, we can look into implementing the spread operator `...` in the expression parser. I like it a lot in Javascript :).
